File: LeetCode/Medium/0005-longest-palindromic-substring/0005-longest-palindromic-substring.java
class Solution {
    public String longestPalindrome(String s) {
        String answer = s.substring(0, 1);

        for(int i = 0; i < s.length() - 1; i++){
            String center = isPalindrome(s, i, i);
            String side = isPalindrome(s, i, i + 1);

            if(center.length() > answer.length()) answer = center;
            if(side.length() > answer.length()) answer = side;
        }

        return answer;
    }

    String isPalindrome(String s, int left, int right){
        while(left >= 0 && right < s.length() && (s.charAt(left) == s.charAt(right))){
            left--;
            right++;
        }

        return s.substring(left + 1, right);
    }
}

File: LeetCode/Medium/0005-longest-palindromic-substring/0005-longest-palindromic-substring.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        string answer;
        int longestSize = 0;

        for(int i = 0 ; i < s.size(); i++){
            int left = i;
            int right = i;

            while(left >= 0 && right < s.size() && (s[left] == s[right])){
                left--;
                right++;
            }

            if(longestSize < right - left + 1){
                answer = s.substr(left + 1, right - left - 1);
                longestSize = right - left + 1;
            }

            left = i;
            right = i + 1;

            while(left >= 0 && right < s.size() && (s[left] == s[right])){
                left--;
                right++;
            }
            
            if(longestSize < right - left + 1){
                answer = s.substr(left + 1, right - left - 1);
                longestSize = right - left + 1;
            }
        }

        return answer;
    }
};